python_modules/dagster/dagster/_core/execution/context/invocation.py

```diff
@@ -206,19 +241,43 @@ def log(self) -> DagsterLogManager:
 
     @property
     def node_handle(self) -> NodeHandle:
-        raise DagsterInvalidPropertyError(_property_msg("solid_handle", "property"))
+        raise DagsterInvalidPropertyError(_property_msg("node_handle", "property"))
 
     @property
-    def solid(self) -> Node:
-        raise DagsterInvalidPropertyError(_property_msg("solid", "property"))
+    def op(self) -> Node:
+        raise DagsterInvalidPropertyError(_property_msg("op", "property"))
 
     @property
     def op_def(self) -> OpDefinition:
-        raise DagsterInvalidPropertyError(_property_msg("op_def", "property"))
+        self.validate_bound_property_access("op_def")
+        assert self._op_def is not None
```

Comment:
I think we shy away from asserts in production code

Reply:
makes sense, we can still use the assert to send a cleaner message though I believe right with the tuple / comma syntax?